### PR TITLE
feature/infra-setup 

### LINKED ISSUE #1: Minimal monorepository setup
Aim is to scaffold basic monorepository structure with minimally needed toolset.

Due to the fact that our project is quite Swifty (Server-Side Swift + Swift-based iOS app), I suggest that we try to stick to tools from Swift ecosystem (swift-pm, swift-log, swift-format & etc.)

As to me, outline of repository structure should like this:

```
server/
client/
shared/
{tools, scripts}/
Package.swift
{configuration_files}.yml
```

We can use Swift Package Manager for automation tasks (like task runners, makefile, taskfile & etc). Also we can give a shot at Swift Package Manaber-based monorepository. For managing git hooks we can use Komondor.

References:
- https://medium.com/reachnow-tech/swift-package-manager-in-a-monorepo-430665228594
- https://developer.apple.com/documentation/swift_packages
- https://github.com/apple/swift-package-manager/tree/master/Documentation
- https://github.com/shibapm/Komondor 
- https://www.lordcodes.com/articles/manage-automation-tasks-using-spm

### LINKED ISSUE #5: Scaffold iOS client core structure
Issue name is pretty descriptive :)

### PR DESCRIPTION
**Pull request by zainoldin**
- Add SPM with dependencies for client & server
- Add SwiftLint, SwiftFormat & R.swift dependencies and their commands in iOS build phase 
- Add iOS client core structure 
- Add iOS Dependencies via SPM

### CODE REVIEW

File: .swiftformat [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# file options
+
+--exclude Fastlane,Pods,Specs,Altel/Resources/Generated,Templates
```
Comment (nugmanoff) [inline .swiftformat]: Fix generated path

File: Client/Teztez/Core/AppDelegate/AppDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+//  Teztez
+//
+//  Created by Almas Zainoldin on 4/18/20.
+//  Copyright © 2020 crococoders. All rights reserved.
+//
+
+import UIKit
+
+@UIApplicationMain
+class AppDelegate: UIResponder, UIApplicationDelegate {
```
Comment (nugmanoff) [inline Client/Teztez/Core/AppDelegate/AppDelegate.swift:12]: Due to the fact that we decided to set deployment target as **13.0** - `AppDelegate` changes to `SceneDelegate`, so we need to adjust Coordinators setup accordingly.

- https://aplus.rs/2020/coordinator-pattern-for-ios13/
- https://markstruzinski.com/2019/08/using-coordinator-with-scene-delegates/

File: Client/Teztez/Core/AppDelegate/AppDelegate.swift [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    var window: UIWindow?
+    private let configurator = AppDelegateConfiguratorFactory.makeDefault()
+
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        setupWindow()
+        _ = configurator.application?(application, didFinishLaunchingWithOptions: launchOptions)
+        return true
+    }
+
+    private func setupWindow() {
```
Comment (nugmanoff) [inline Client/Teztez/Core/AppDelegate/AppDelegate.swift]: Also it'd be nice if we can finally move this setup to `StartupAppDelegateConfigurator` (the thing me & Nurlan discussed). Please try doing it

File: Client/Teztez/Core/Architecture/Navigation/ApplicationCoordinator.swift [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import Foundation
+
+public final class ApplicationCoordinator: ParentCoordinator {
+    private let instructor: ApplicationLaunchInstructor
+    private let coordinatorFactory: CoordinatorFactory
+
+    init(instructor: ApplicationLaunchInstructor,
+         router: Router,
+         coordinatorFactory: CoordinatorFactory,
+         notificationCenter: NotificationCenter = .default) {
```
Comment (nugmanoff) [inline Client/Teztez/Core/Architecture/Navigation/ApplicationCoordinator.swift]: I think we can remove this for now

File: Client/Teztez/Core/Design System/FontStyles.swift [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+//  Teztez
+//
+//  Created by Almas Zainoldin on 4/19/20.
+//  Copyright © 2020 crococoders. All rights reserved.
+//
+
+import UIKit
+
+extension UIFont {
+    // MARK: Common Font Styles
```
Comment (nugmanoff) [inline Client/Teztez/Core/Design System/FontStyles.swift]: Redundant mark comment

File: Client/Teztez/Core/Language.swift [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+//  Copyright © 2020 crococoders. All rights reserved.
+//
+
+import UIKit
+
+enum Language: String, CaseIterable {
+    case eng
+}
+
+extension Language {
```
Comment (nugmanoff) [inline Client/Teztez/Core/Language.swift:15]: I think we can do better and use some framework or better solution to deal with Localization. 

Our app is going to have quite nice UX and changing the language should also be nice and be updated "on the fly" (realtime). The same way Telegram does it.

But I believe that our current implementation is not capable of doing that. (You implement localization as a separate feature)

File: Client/Teztez.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -0,0 +1,882 @@
...
+					"DEBUG=1",
+					"$(inherited)",
+				);
+				GCC_WARN_64_TO_32_BIT_CONVERSION = YES;
+				GCC_WARN_ABOUT_RETURN_TYPE = YES_ERROR;
+				GCC_WARN_UNDECLARED_SELECTOR = YES;
+				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
+				GCC_WARN_UNUSED_FUNCTION = YES;
+				GCC_WARN_UNUSED_VARIABLE = YES;
+				IPHONEOS_DEPLOYMENT_TARGET = 11.0;
```
Comment (nugmanoff) [inline Client/Teztez.xcodeproj/project.pbxproj]: This one should be 13.0

**Review by nugmanoff — CHANGES_REQUESTED** [review]

File: Client/Teztez/Core/Language.swift (hunk 6 above)
Comment (zainoldin) [reply Client/Teztez/Core/Language.swift:15]: I think we should create new issue and do all localization logic in it.

**Review by nugmanoff — APPROVED** [review]

**Pull request merged**